SU2_CFD/src/drivers/CDiscAdjMultizoneDriver.cpp

```diff
@@ -941,7 +940,9 @@ void CDiscAdjMultizoneDriver::SetResidual_BGS(unsigned short iZone) {
 
   for (unsigned short iSol=0; iSol < MAX_SOLS; iSol++) {
     auto solver = solver_container[iZone][INST_0][MESH_0][iSol];
-    if (solver && solver->GetAdjoint())
+    if (solver && solver->GetAdjoint()) {
       solver->ComputeResidual_Multizone(geometry_container[iZone][INST_0][MESH_0], config_container[iZone]);
+      solver->GetNodes()->Set_BGSSolutionExtra_k();
+    }
```

Comment:
This is not super straight forward as `CDiscAdjMultizoneDriver::Set_BGSSolution_k_To_Solution`  is a subset of what `ComputeResidual_Multizone` does. `ComputeResidual_Multizone` is used in the primal and adjoint so if the preferred way is to cut the `Set_BGSSolution_k_To_Solution` out, then it needs to be taken care of for the primal as well

Reply:
lets see whether [ef6bbc4](https://github.com/su2code/SU2/pull/1536/commits/ef6bbc43bb29586df2c0a11a95a931b05693b997) works